File: 5-delete-data.py
import sqlite3

conn = sqlite3.connect('database.db')
cursor = conn.cursor()

print('🗑️ Deleting todos...')
print()

# ก่อนลบ
print('Before delete:')
cursor.execute('SELECT COUNT(*) as total FROM todos')
before = cursor.fetchone()[0]
print(f"  Total todos: {before}")
print()

# ลบ todo ที่ id = 5
cursor.execute('DELETE FROM todos WHERE id = ?', (5,))
changes = cursor.rowcount
conn.commit()

print(f"✅ Deleted {changes} todo")
print()

# หลังลบ
print('After delete:')
cursor.execute('SELECT COUNT(*) as total FROM todos')
after = cursor.fetchone()[0]
print(f"  Total todos: {after}")
print()

# ดูข้อมูลที่เหลือ
print('Remaining todos:')
cursor.execute('SELECT * FROM todos')
remaining = cursor.fetchall()
print(f"{'ID':>4} {'Task':<30} {'Done':>6}")
print('-' * 45)
for todo in remaining:
    print(f"{todo[0]:>4} {todo[1]:<30} {todo[2]:>6}")

print()
conn.close()

File: Week2-day4/Workshop 13-SQLite Basics/Level 1-Todo App Database/5-delete-data.js
const Database = require('better-sqlite3');
const db = new Database('database.db');

console.log('🗑️ Deleting todos...');
console.log('');

// ก่อนลบ
console.log('Before delete:');
const before = db.prepare('SELECT COUNT(*) as total FROM todos').get();
console.log(`Total todos: ${before.total}`);
console.log('');

// ลบ todo ที่ id = 5
const deleteTodo = db.prepare('DELETE FROM todos WHERE id = ?');
const result = deleteTodo.run(5);

console.log(`✅ Deleted ${result.changes} todo`);
console.log('');

// หลังลบ
console.log('After delete:');
const after = db.prepare('SELECT COUNT(*) as total FROM todos').get();
console.log(`Total todos: ${after.total}`);
console.log('');

// ดูข้อมูลที่เหลือ
console.log('Remaining todos:');
const remaining = db.prepare('SELECT * FROM todos').all();
console.table(remaining);

console.log('');
db.close();
